#!/usr/bin/env python
PACKAGE = "softdrone_core"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("q_pos", double_t, 0, "Position penalty", .001, 0, 10000)
gen.add("q_vel", double_t, 0, "Velocity penalty", .001, 0, 10000)
gen.add("q_acc", double_t, 0, "Acceleration penalty", .001, 0, 10000)
gen.add("q_jerk", double_t, 0, "Jerk penalty", .001, 0, 10000)
gen.add("q_snap", double_t, 0, "Snap penalty", .001, 0, 10000)
gen.add("extra_q_reg", double_t, 0, "Q regularization", .1, 0, 100)
gen.add("k_t_scale", double_t, 0, "k_t scale", 100, 0, 10000)
gen.add("j_scale", double_t, 0, "J scale", 1, 0, 1000)
gen.add("t_grasp_hardcode", double_t, 0, "t grasp unscaled", 0.5, 0, 1)

exit(gen.generate(PACKAGE, "polynomial_planner_node", "PolynomialPlanner"))
